model/task/task.go

```diff
@@ -3254,9 +3284,48 @@ func (t *Task) GetFailedTestSample(ctx context.Context, env evergreen.Environmen
 		return nil, nil
 	}
 
+	flags, err := evergreen.GetServiceFlags(ctx)
+	if err != nil {
+		return []string{}, errors.Wrap(err, "retrieving service flags")
+	}
+	if !flags.EvergreenTestResultsDisabled {
+		output, ok := t.getTaskOutputSafe()
+		if !ok {
+			return []string{}, nil
+		}
+		if len(taskOpts) == 0 {
+			return []string{}, nil
+		}
+		return output.TestResults.GetMergedFailedTestSample(ctx, taskOpts)
+	}
+
 	return testresult.GetMergedFailedTestSample(ctx, env, taskOpts)
 }
 
+// GetFailedTestSamples returns failed test samples filtered as specified by
+// the optional regex filters for each task specified.
+func (t *Task) GetFailedTestSamples(ctx context.Context, env evergreen.Environment, taskOpts []testresult.TaskOptions, regexFilters []string) ([]testresult.TaskTestResultsFailedSample, error) {
```

Comment:
This only needs to be implemented in the `taskoutput` package. The functions on the task struct are meant to be convenience functions for a single task, this function fetches failed test samples for an arbitrary amount of tasks.

Reply:
Removed this from the task struct functions